src/lib/VTXSPI/devVTXSPI.cpp

```diff
@@ -85,7 +85,22 @@ static void rtc6705WriteRegister(uint32_t regData)
     }
 
     #if defined(PLATFORM_ESP32)
-        vtxSPI->transferBits(regData, nullptr, 25);
+        // On some ESP32 MCUs there's a silicon bug which affects all 8n+1 bit and 1 bit transfers where the 
+        // last bit sent is corrupt.
+        // See: https://github.com/ExpressLRS/ExpressLRS/pull/2406#issuecomment-1722573356
+        //
+        // To reproduce, use an ESP32S3 and 25 bit transfers, change from channel A4 to A1, then A1 to A4 (ok), 
+        // then A4 to A3, then A3 to A4 (fail)
+        //
+        // 12816, 9286833 appears on the scope when changing from A:1->A:4
+        // 12816, 26064049 appears on the scope when changing from A:3->A:4
+        // 
+        // 9286833  = 0_1000_1101_1011_0100_1011_0001
+        // 26064049 = 1_1000_1101_1011_0100_1011_0001
+        //
+        // So, to workaround this issue, we send 32 bits, and the RTC67005 ignores the last 7 bits.
+        regData |= 0b1111111 << 25;
+        vtxSPI->transferBits(regData, nullptr, 32);
     #else
```

Comment:
Remove these

Reply:
Change the last comment line and the transfer function so it becomes
```cpp
    // Since the RTC6705 just ignores the extra bits, we send 32 bits and the RTC6705 ignores the last 7 bits.
    vtxSPI->transfer32(regData);
```

Reply:
done as requested.  Note that I've not tested this change here in any way so I'm relying on your expertise here @pkendall64

Reply:
Thanks, but you didn't remove the `#if` and the entire `#else` section as it's not required.

Reply:
updated again, compiles ok here for ESP32, ESP32S3 and ESP8285.